### PR TITLE
fix no result reference

### PR DESCRIPTION
**Pull request by alavieille**

### CODE REVIEW

**Comment by ngilain** [comment]
:+1:

**Comment by nicolasThal** [comment]
:+1:

**Comment by itkg-alavieille** [comment]
@itkg-ngilain @nicolasThal i refacto this PR

**Comment by nicolasThal** [comment]
I think you should update the tests

**Comment by nicolasThal** [comment]
:+1: for the refacto

**Comment by alavieille** [comment]
@nicolasThal, @itkg-ngilain @itkg-nanne

**Comment by nicolasThal** [comment]
:+1:

File: Pagination/Tests/MongoTrait/FilterTypeStrategy/Strategies/ReferenceFilterStrategyTest.php [hunk 1]
```diff
@@ -97,15 +97,19 @@ public function testGenerateFilterWithNoPaginationRepsitory()
...
+     * @param int|null $countOrFilter
      *
      * @dataProvider provideGenerateFilter
      */
-    public function testGenerateFilter($columnsTree, $expectedFilter)
+    public function testGenerateFilter($columnsTree, $countOrFilter = null)
     {
         $filter = $this->strategy->generateFilter($columnsTree, $this->value, $this->documentName);
-        $this->assertEquals($expectedFilter, $filter);
+        if (null === $countOrFilter) {
```
Comment (ngilain) [inline Pagination/Tests/MongoTrait/FilterTypeStrategy/Strategies/ReferenceFilterStrategyTest.php:108]: Can you put the default value to 0 ? It's weird that a count has null value

Comment (alavieille) [inline Pagination/Tests/MongoTrait/FilterTypeStrategy/Strategies/ReferenceFilterStrategyTest.php:108]: null indicates that there is no filter created.  not that there is a filter with no correspondence

Comment (alavieille) [inline Pagination/Tests/MongoTrait/FilterTypeStrategy/Strategies/ReferenceFilterStrategyTest.php:108]: @itkg-ngilain

Comment (ngilain) [inline Pagination/Tests/MongoTrait/FilterTypeStrategy/Strategies/ReferenceFilterStrategyTest.php:108]: ok then !

**Comment by ngilain** [comment]
:+1:

**Pull request merged**